### DISCUSSION TITLE
Is it possible to add groups dynamically in stencil ?

### DISCUSSION THREAD

**Discussion by TomPast** [clientIO/joint#2383] [Q&A] [answered]
### Introduction

Hello,
How can I add groups dynamically after stencil creation without creating a new stencil ?

For example : 
```
const stencil = new ui.Stencil({
    paper: scroller,
    usePaperGrid: true,
    width: 300,
    snaplines,
    paperOptions: () => ({
      model: new dia.Graph({}, { cellNamespace: shapes }),
      cellViewNamespace: shapes,
    }),
  });

stencil.render();
stencilElement?.current.appendChild(stencil.el);


//I would like to do something like this 
stencil.groups.load({
  standard: {
    name: "Standard",
    index: 0,
    label: "Standard",
    closed: false,
  },
})
```

Thanks in advance!

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment, marked as answer]
There is no API to add/remove/change groups at this moment. You can re-render the stencil though as shown in https://github.com/clientIO/joint/discussions/2226#discussioncomment-6241616.
